v4/export/sql.go

```diff
@@ -277,3 +325,118 @@ func simpleQuery(db *sql.DB, sql string, handleOneRow func(*sql.Rows) error) err
 	}
 	return nil
 }
+
+func pickupPossibleField(dbName, tableName string, db *sql.DB, conf *Config) (string, error) {
+	// If detected server is TiDB, try using _tidb_rowid
+	if conf.ServerInfo.ServerType == ServerTypeTiDB {
+		ok, err := SelectTiDBRowID(db, dbName, tableName)
+		if err != nil {
+			return "", nil
+		}
+		if ok {
+			return "_tidb_rowid", nil
+		}
+	}
+	// try to use pk
+	fieldName, err := GetPrimaryKeyName(db, dbName, tableName)
+	if err != nil {
+		return "", err
+	}
+	// try to use first uniqueIndex
+	if fieldName == "" {
+		fieldName, err = GetUniqueIndexName(db, dbName, tableName)
+	}
+
+	// there is no proper index
+	if fieldName == "" {
+		return "", nil
+	}
+
+	query := fmt.Sprintf("SELECT DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS "+
+		"WHERE table_name = '%s' AND COLUMN_NAME = '%s'", tableName, fieldName)
+	var fieldType string
+	handleOneRow := func(rows *sql.Rows) error {
```

Comment:
If we only query one row, may we can try to use `db.QueryRow`?

Reply:
fixed